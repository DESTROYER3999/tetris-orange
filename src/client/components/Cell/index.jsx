import { connect } from "react-redux";
import CellSub from "./subcomponent";

const Cell = connect()(CellSub);

export default Cell;
