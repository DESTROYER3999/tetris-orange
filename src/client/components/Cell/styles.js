import styled, { css } from "styled-components";

export const StyledContainer = styled.div`
  flex: 1 10%;
  border: 1px solid black;
  ${props =>
    props.color &&
    !props.isSpectre &&
    css`
      border: 5px outset ${props => props.color};
      background-color: ${props => props.color};
    `};
  ${props =>
    props.isSpectre &&
    css`
      border: 0px;
      ${props =>
        props.color &&
        css`
          background: #ff8a0080;
        `};
    `};
`;

export const StyledLetter = styled.p`
  font-size: 20px;
  display: flex;
  justify-content: center;
  align-items: center;
  transition: font-size 0.2;
  &:hover {
    font-size: 30px;
  }
`;
