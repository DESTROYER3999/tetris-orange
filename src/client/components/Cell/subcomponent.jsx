import React from "react";
import PropTypes from "prop-types";
import { StyledContainer, StyledLetter } from "./styles";

const CellSub = ({ color, isSpectre = false, letter = "" }) => {
  return (
    <StyledContainer color={color} isSpectre={isSpectre}>
      <StyledLetter>{letter}</StyledLetter>
    </StyledContainer>
  );
};

CellSub.propTypes = {
  color: PropTypes.string.isRequired
};

export default CellSub;
