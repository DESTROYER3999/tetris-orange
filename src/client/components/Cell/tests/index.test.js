import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import configureStore from "redux-mock-store";
import Cell from "../index";

const mockStore = configureStore();

const store = mockStore({});

describe("<Cell />", () => {
  test("Cell rendering with store without crashing", () => {
    const wrapper = shallow(<Cell color="" store={store} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
