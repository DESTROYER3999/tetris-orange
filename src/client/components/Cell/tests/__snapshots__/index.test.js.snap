// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cell /> Cell rendering with store without crashing 1`] = `
<CellSub
  color=""
  dispatch={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
